### PR TITLE
Merging of ``use-implicit-booleaness``-like checkers for consistency and performance

### LINKED ISSUE #5234: C1901 (compare-to-empty-string) message is not explicit about the problem and the solution
### Bug description

Consider

```python
   val_type, val_value = get_type(name)  # returns str, Any
   if val_type is Environment:
     if val_value is None or val_value is "":  # or ==
```

The last line produces a pylint error, with the common guidance (e.g https://vald-phoenix.github.io/pylint-errors/plerr/errors/compare-to-empty-string/C1901.html) suggesting that you should use:

```python
    if val_value is None or not len(val_value):
```

The user used `val_value is ""` to explicitly say: is it a string, and is it empty?

To avoid doing this `is` or `==` comparison, the user would need to write:

```python
    if val_value is None or (isinstance(val_value, str) and not len(val_value)):
```

example of following the guidance too literally:

```python
    def fn(v):
      if len(v):
          return "empty"
      return "other"

    fn(None)    # OK: returns "other"
    fn("a")     # OK: returns "other"
    fn([None])  # OK: returns "other"
    fn("")      # OK: returns "empty"
    fn([])      # WRONG: returns "other"
    fn(0)       # ERROR: int has no len
```

### Configuration

_No response_

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
a.py.3.1: [C1901(compare-to-empty-string), a.py] Avoid comparisons to empty string [229,1]
```

### Expected behavior

This should either be disabled by default or it should provide better guidance. Why is `x is ""` bad?

### Pylint version

```shell
pylint 1.9.5,
astroid 1.6.6
Python 2.7.18 (v2.7.18:8d21aa21f2, Apr 20 2020, 13:25:05) [MSC v.1500 64 bit (AMD64)]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### LINKED ISSUE #6871: Uniformize ``compare-to-zero``, ``compare-to-empty-string`` with the ``use-implicit-booleaness-x`` checks
### Current problem

 ``compare-to-zero`` and ``compare-to-empty-string`` are the same check basically than ``use-implicit-booleaness-not-copmparison``.  

Doing ``x != 0`` feel safer and easier to understand than ``bool(x)`` for int (you need a background in C to understand that). So``compare-to-zero`` feel more opinionated  which is probably why they are in their own optional checker. 

``compare-to-empty-string`` though is really an ``use-implicit-booleaness-not-copmparison`` check.  

### Desired solution

I think it would make sense to merge ``compare-to-empty-string`` . Renaming them so they all follow the same pattern could be an option.

### Additional context

https://github.com/PyCQA/pylint/pull/6861

I've done the work in #6870 then realized that ``compare-to-zero`` was opinionated.

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Follow-up to #8630 
Closes #6871
Closes #5234

### CODE REVIEW

File: pylint/checkers/refactoring/implicit_booleaness_checker.py [hunk 1]
```diff
@@ -4,13 +4,21 @@
...
+
 import astroid
 from astroid import bases, nodes
 
 from pylint import checkers
 from pylint.checkers import utils
 
 
+def _is_constant_zero(node):
+    return isinstance(node, astroid.Const) and node.value == 0
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/refactoring/implicit_booleaness_checker.py:19]: This trigger the check which made me think it's opinionated.

File: pylint/checkers/refactoring/implicit_booleaness_checker.py [hunk 2]
```diff
@@ -139,6 +152,80 @@ def visit_unaryop(self, node: nodes.UnaryOp) -> None:
     @utils.only_required_for_messages("use-implicit-booleaness-not-comparison")
     def visit_compare(self, node: nodes.Compare) -> None:
         self._check_use_implicit_booleaness_not_comparison(node)
+        self._check_compare_to_empty_string(node)
+        self._check_compare_to_zero(node)
+
+    def _check_compare_to_zero(self, node: nodes.Compare) -> None:
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/refactoring/implicit_booleaness_checker.py:158]: Those two functions are highly duplicated, they could be merged if we decide that everything can be merged in this checker.

**Comment by Pierre-Sassoulas** [comment]
Following #7629 we can merge everything in the same checker but still make one check optional by default.

**Comment by Pierre-Sassoulas** [comment]
There might be some optimization between ``"use-implicit-booleaness-not-comparison`` and the two other but they're not obvious, the logic of "to-string" and "to-zero" would need a serious refactor. I think it's good enough for now.

File: pylint/checkers/refactoring/implicit_booleaness_checker.py (hunk 2 above)
Comment (Pierre-Sassoulas) [reply pylint/checkers/refactoring/implicit_booleaness_checker.py]: I did that.

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**